src/modules/auth/domain/auth.service.ts

```diff
@@ -0,0 +1,68 @@
+import { ClassProvider, Inject, Injectable, HttpException, HttpStatus } from "@nestjs/common";
+import {
+	ServiceLoginInput,
+	ServiceCheckTokenInput,
+	ServiceAuth,
+	AuthServiceDefinition,
+	AuthServiceName,
+} from "src/modules/auth/domain/auth.service.types";
+import {
+	RepoAuth,
+	AuthRepositoryDefinition,
+	AuthRepositoryName,
+} from "src/modules/auth/repository/auth.repository.types";
+import CryptoJS = require("crypto-js"); // hash
+import crypto = require("crypto"); // token
+
+const VALID_TOKEN_TIME: bigint = 2628n * 10000n; // Amount of seconds in a mounth
+export const SALT = "3b7f9e2c1d4a5e6b7c8f0a1d9e3b2c4d";
+
+@Injectable()
+export class AuthService implements AuthServiceDefinition {
+	constructor(@Inject(AuthRepositoryName) private readonly authRepository: AuthRepositoryDefinition) {}
+
+	public static Provider(): ClassProvider<AuthService> {
+		return {
+			provide: AuthServiceName,
+			useClass: AuthService,
+		};
+	}
+
+	async login({ username, password }: ServiceLoginInput): Promise<ServiceAuth> {
```

Comment:
Why are you defining username and loginHash? You aren't actually using them. If it's not required in the function why would you define it as an input? Remove this. Or change up your function to use it becauase `username` seems like something we really need